s2spy/_base_calendar.py

```diff
@@ -81,24 +86,24 @@ def _parse_anchor(self, anchor_str: str) -> Tuple[str, str]:
             fmt = "%m"
         else:
             raise ValueError(
-                f"Anchor input '{anchor_str}' does not match expected format"
+                f"Anchor input '{anchor_str}' does not match expected format."
             )
         return anchor_str, fmt
 
     def _append(self, period_block):
         """Append target/precursor periods to the calendar."""
-        if period_block.target:
+        if period_block._target:
             self._targets.append(period_block)
             # count length
             self._total_length_target += (
-                period_block.length.kwds["days"] + period_block.gap.kwds["days"]
+                period_block._length.kwds["days"] + period_block._gap.kwds["days"]
```

Comment:
Why did you change `length` to `_length` here (same for `gap`). These are still accessible as their are properties, so there is no need to use the `_` versions.

Reply:
Hmmm...Good point. I didn't think about it. Normally `length` is used for display and `_length` is used internally, as there are always some further processing related to these protected values. But indeed it is not the case here. I will access them via properties. Thanks for the comment 💯 .